class.jetpack-autoupdate.php

```diff
@@ -125,7 +180,30 @@ function save_update_data() {
 			}
 		}
 
+		// Only show them if the user has the ability to update them.
+		// Check for permission.
+		if (  ! current_user_can( 'update_plugins' ) ) {
```

Comment:
I would remove this checks and always save the entire `$updates` array, if a user without the privileges fires this it will override the Jetpack_Option and sync a false 0 value to wpcom.